### PR TITLE
feat: add dms rocketmq user resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add dms rocketMQ user resource, this resouce can be user to manage the user of rocketMQ
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add dms rocketMQ user resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dms/' TESTARGS='-run TestAccDmsRocketMQUser_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dms/ -v -run TestAccDmsRocketMQUser_basic -timeout 360m -parallel 4       
=== RUN   TestAccDmsRocketMQUser_basic
=== PAUSE TestAccDmsRocketMQUser_basic
=== CONT  TestAccDmsRocketMQUser_basic
--- PASS: TestAccDmsRocketMQUser_basic (17.24s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dms       17.317s
```

### CODE REVIEW

File: docs/resources/dms_rocketmq_user.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `instance_id` - (Required, String, ForceNew)
```
Comment (Lance52259) [inline docs/resources/dms_rocketmq_user.md]: the description is missing.

File: docs/resources/dms_rocketmq_user.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  Value options: **PUB|SUB**, **PUB**, **SUB**, **DENY**.
+
+* `default_group_perm` - (Optional, String) Specifies the default consumer group permissions.
+  Value options: **PUB|SUB**, **PUB**, **SUB**, **DENY**.
+
+* `topic_perms` - (Optional, List) Specifies the special topic permissions.
+  The [PermsRef](#DmsRocketMQUser_PermsRef) structure is documented below.
+
+* `group_perms` - (Optional, List) Specifies the special consumer group permissions.
+  The [PermsRef](#DmsRocketMQUser_PermsRef) structure is documented below.
```
Comment (Lance52259) [inline docs/resources/dms_rocketmq_user.md]:
```suggestion
* `topic_perms` - (Optional, List) Specifies the special topic permissions.
  The [permission](#DmsRocketMQUser_PermsRef) structure is documented below.

* `group_perms` - (Optional, List) Specifies the special consumer group permissions.
  The [permission](#DmsRocketMQUser_PermsRef) structure is documented below.
```

File: docs/resources/dms_rocketmq_user.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+* `perm` - (Optional, String) Indicates the permissions of the topic or consumer group.
+  Value options: **PUB|SUB**, **PUB**, **SUB**, **DENY**.
+
+<a name="DmsRocketMQUser_PermsRef"></a>
+The `PermsRef` block supports:
+
+* `name` - (Optional, String) Indicates the name of a topic or consumer group.
+
+* `perm` - (Optional, String) Indicates the permissions of the topic or consumer group.
+  Value options: **PUB|SUB**, **PUB**, **SUB**, **DENY**.
```
Comment (Lance52259) [inline docs/resources/dms_rocketmq_user.md]:
```suggestion
<a name="DmsRocketMQUser_PermsRef"></a>
The `topic_perms` and `group_perms` block supports:

* `name` - (Optional, String) Indicates the name of a topic or consumer group.

* `perm` - (Optional, String) Indicates the permissions of the topic or consumer group.
  Value options: **PUB|SUB**, **PUB**, **SUB**, **DENY**.
```
define a structure just ok.

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_user.go [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+		deleteRocketmqUserProduct = "dms"
+	)
+	deleteRocketmqUserClient, err := config.NewServiceClient(deleteRocketmqUserProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DmsRocketMQUser Client: %s", err)
+	}
+
+	parts := strings.SplitN(d.Id(), "/", 2)
+	if len(parts) != 2 {
+		return diag.Errorf("invalid id format, must be <instance_id>/<topic>")
```
Comment (Lance52259) [inline huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_user.go]: topic?

File: docs/resources/dms_rocketmq_user.md (hunk 1 above)
Comment (houpeng80) [reply docs/resources/dms_rocketmq_user.md]: done

File: docs/resources/dms_rocketmq_user.md (hunk 2 above)
Comment (houpeng80) [reply docs/resources/dms_rocketmq_user.md]: done

File: docs/resources/dms_rocketmq_user.md (hunk 3 above)
Comment (houpeng80) [reply docs/resources/dms_rocketmq_user.md]: done

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_user.go (hunk 4 above)
Comment (houpeng80) [reply huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_user.go]: it has been changed to user

File: docs/resources/dms_rocketmq_user.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+## Import
+
+The rocketmq user can be imported using the rocketMQ instance ID and user access key separated by a slash, e.g.
+
+```
+$ terraform import huaweicloud_dms_rocketmq_user.test c8057fe5-23a8-46ef-ad83-c0055b4e0c5c/user_serurity_key
```
Comment (hmbig2) [inline docs/resources/dms_rocketmq_user.md]:
```suggestion
$ terraform import huaweicloud_dms_rocketmq_user.test c8057fe5-23a8-46ef-ad83-c0055b4e0c5c/access_key

```

Comment (houpeng80) [reply docs/resources/dms_rocketmq_user.md]: done

File: docs/resources/dms_rocketmq_user.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+* `white_remote_address` - (Optional, String) Specifies the IP address whitelist.
+
+* `admin` - (Optional, Bool) Specifies whether the user is an administrator.
+
+* `default_topic_perm` - (Optional, String) Specifies the default topic permissions.
+  Value options: **PUB|SUB**, **PUB**, **SUB**, **DENY**.
+
+* `default_group_perm` - (Optional, String) Specifies the default consumer group permissions.
+  Value options: **PUB|SUB**, **PUB**, **SUB**, **DENY**.
```
Comment (Lance52259) [inline docs/resources/dms_rocketmq_user.md:59]: please add the description of the default value.

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rocketmq_user_test.go [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+		getRocketmqUserProduct = "dms"
+	)
+	getRocketmqUserClient, err := config.NewServiceClient(getRocketmqUserProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating DmsRocketMQUser Client: %s", err)
+	}
+
+	parts := strings.SplitN(state.Primary.ID, "/", 2)
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid id format, must be <instance_id>/<user>")
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rocketmq_user_test.go]: user or access key?

File: huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_user.go [hunk 8]
```diff
@@ -0,0 +1,389 @@
...
+		deleteRocketmqUserProduct = "dms"
+	)
+	deleteRocketmqUserClient, err := config.NewServiceClient(deleteRocketmqUserProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DmsRocketMQUser Client: %s", err)
+	}
+
+	parts := strings.SplitN(d.Id(), "/", 2)
+	if len(parts) != 2 {
+		return diag.Errorf("invalid id format, must be <instance_id>/<user>")
```
Comment (Lance52259) [inline huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_user.go]: ditto

Comment (houpeng80) [reply huaweicloud/services/dms/resource_huaweicloud_dms_rocketmq_user.go]: done

File: huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rocketmq_user_test.go (hunk 7 above)
Comment (houpeng80) [reply huaweicloud/services/acceptance/dms/resource_huaweicloud_dms_rocketmq_user_test.go]: done

File: docs/resources/dms_rocketmq_user.md (hunk 6 above)
Comment (houpeng80) [reply docs/resources/dms_rocketmq_user.md:59]: done

**Pull request merged**